src/main/steps/respondent/consent-to-application/ConsentMapper.ts

```diff
@@ -1,59 +1,42 @@
-  let consentFromResponsent: Consent;
-  let reasonForNotConsenting = '';
-  let courtOrderDetails = '';
-  if (req.session.userCase.doYouConsent === YesOrNo.NO) {
-    reasonForNotConsenting = req.session.userCase.reasonForNotConsenting!;
-  }
-  if (req.session.userCase.courtPermission === YesOrNo.YES) {
-    courtOrderDetails = req.session.userCase.courtOrderDetails!;
+
+export const prepareRequest = (req: CaseWithId): Consent => {
+  const { doYouConsent, courtPermission, reasonForNotConsenting, courtOrderDetails } = req;
+
+  const request: Consent = {};
+
+  Object.assign(request, {
+    consentToTheApplication: doYouConsent,
+    permissionFromCourt: courtPermission,
+    noConsentReason: reasonForNotConsenting,
+    courtOrderDetails,
+  });
+
+  // data cleanup
+  if (doYouConsent === YesOrNo.YES) {
+    delete request.noConsentReason;
   }
-  if (respondent?.value?.response && respondent?.value?.response?.consent) {
-    consentFromResponsent = respondent?.value?.response?.consent;
-    consentFromResponsent.consentToTheApplication = req.session.userCase.doYouConsent;
-    consentFromResponsent.noConsentReason = reasonForNotConsenting;
-    consentFromResponsent.applicationReceivedDate = toApiDate(req.session.userCase.applicationReceivedDate);
-    consentFromResponsent.permissionFromCourt = req.session.userCase.courtPermission;
-    consentFromResponsent.courtOrderDetails = courtOrderDetails;
-    respondent.value.response.consent = consentFromResponsent;
-  } else {
-    respondent.value.response = {
-      consent: {
-        consentToTheApplication: req.session.userCase.doYouConsent,
-        noConsentReason: reasonForNotConsenting,
-        applicationReceivedDate: toApiDate(req.session.userCase.applicationReceivedDate),
-        permissionFromCourt: req.session.userCase.courtPermission,
-        courtOrderDetails,
-      },
-    };
+
+  if (courtPermission === YesOrNo.NO) {
+    delete request.courtOrderDetails;
   }
-  return respondent;
+
+  return request;
 };
 
-export const getConsentDetails = (respondent: Respondent, req: AppRequest): Partial<CaseWithId> => {
-  if (respondent?.value?.response?.consent?.consentToTheApplication === YesOrNo.NO) {
-    req.session.userCase.doYouConsent = YesOrNo.NO;
-    req.session.userCase.reasonForNotConsenting = respondent?.value?.response?.consent.noConsentReason;
-  } else {
-    req.session.userCase.doYouConsent = YesOrNo.YES;
-    req.session.userCase.reasonForNotConsenting = '';
-  }
-  if (respondent?.value?.response?.consent?.permissionFromCourt === YesOrNo.NO) {
-    req.session.userCase.courtPermission = YesOrNo.NO;
-    req.session.userCase.courtOrderDetails = '';
-  } else {
-    req.session.userCase.courtPermission = YesOrNo.YES;
-    req.session.userCase.courtOrderDetails = respondent?.value?.response?.consent?.courtOrderDetails;
-  }
-  req.session.userCase.applicationReceivedDate = fromApiDate(
```

Comment:
Missing.

Reply:
Added `applicationReceivedDate` to mapDetails method as well